components/event-bus/internal/knative/util/kn-eventing.go

```diff
@@ -336,12 +336,12 @@ func resendMessage(httpClient *http.Client, channel *messagingV1Alpha1.Channel,
 	return nil
 }
 
-func makeChannel(name string, namespace string, labels *map[string]string) *messagingV1Alpha1.Channel {
+func makeChannel(prefix string, namespace string, labels *map[string]string) *messagingV1Alpha1.Channel {
 	return &messagingV1Alpha1.Channel{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: namespace,
-			Name:      name,
-			Labels:    *labels,
+			Namespace:    namespace,
+			GenerateName: prefix,
```

Comment:
```suggestion
			GenerateName: prefix+"-",
```
(or include it in all callers)

Reply:
there's a comples validation for adding suffix

Reply:
Can you elaborate?

Reply:
https://github.com/kyma-project/kyma/pull/5704/files#diff-e45ab281cd2559e951efa6376363324bR344

Reply:
some external systems have `.` char in the eventType, so m ensuring i get rid of all the special chars before i send the GenerateName to kubernetes. and after cleaning the string i add the "-" suffix

Reply:
Which is fine, since you do it after the replacement, or do I get it wrong?

Reply:
yes, we add the suffix after replacement